generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/generateGraphQLType.kt

```diff
@@ -63,7 +65,15 @@ private fun objectFromReflection(generator: SchemaGenerator, type: KType, typeIn
 
     return generator.cache.buildIfNotUnderConstruction(kClass, typeInfo) {
         val graphQLType = getGraphQLType(generator, kClass, type, typeInfo)
-        generator.config.hooks.willAddGraphQLTypeToSchema(type, graphQLType)
+
+        /*
+         * For a field using the meta union annotation, the `type` is `Any`, but we need to pass the annotation with the meta union annotation as the type
+         * since that is really the type generated from reflection and has any potential directives on it needed by the hook
+         */
+        val metaUnion: Annotation? = typeInfo.fieldAnnotations.firstOrNull { it.getMetaUnionAnnotation() != null }
+        val resolvedType = if (kClass.isInstance(Any::class) && metaUnion != null) metaUnion.annotationClass.createType() else type
```

Comment:
brackets please

Reply:
this should have been caught by our lint checker?